phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -550,15 +550,48 @@ public BaseResultIterators(QueryPlan plan, Integer perScanLimit, Integer offset,
         }
         return ranges;
     }
-    
+
+    private List<KeyRange> getRegionRowKeyRanges() throws SQLException {
```

Comment:
I might be missing some context here but I see this piece of code in this file which can re-used and handles the required corner cases.
```
    static final Function<HRegionLocation, KeyRange> TO_KEY_RANGE = new Function<HRegionLocation, KeyRange>() {
        @Override
        public KeyRange apply(HRegionLocation region) {
            return KeyRange.getKeyRange(region.getRegionInfo().getStartKey(), region.getRegionInfo().getEndKey());
        }
    };
```

Reply:
Can you add some comments here as to how this method will help?

Reply:
Added comments and removed the TO_KEY_RANGE dead code.